gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -342,8 +412,24 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
       }
     }
 
+    // Compare the modification timestamp of the spec being added if the scheduler is being initialized, ideally we
+    // don't even want to do the same update twice as it will kill the existing flow and reschedule it unnecessarily
+    Long modificationTime = Long.valueOf(flowSpec.getConfigAsProperties().getProperty(MysqlBaseSpecStore.modificationTimeKey, "0"));
+    if (this.isSchedulingSpecsFromCatalog()) {
```

Comment:
we can do this compare all the time even if it's not the loading time period? What's the concern for not doing that?

Reply:
If not scheduling period, then as of now we should always be processing most recent update but we can do the check anyway since its small.